### ISSUE TITLE
Block toolbar mover buttons trigger a focus loss

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#58511]
### Description

This appears to be a regression, although I'm not sure when it was introduced.

When using the keyboard to activate the 'Move up' and 'Move down' buttons in the block toolbar, focus is expected to stay on the button that was activated. Instead, there's now a focus loss and focus is moved to the first button within the toolbar.

I haven't dive into the code but I suspect a full-rerendering occurs and the toolbar sets focus on its first focusable control each time.

This unexpected behavior doesn't happen on previous WordPress version, as such it appears to be a regression that should be fixed in the next release. Cc @annezazu 

Animated GIF to illustrate:

![movers](https://github.com/WordPress/gutenberg/assets/1682452/aee2a6ba-4f5e-4678-930a-920a0b67401c)

### Step-by-step reproduction instructions

- Create a post with many blocks.
- Select a paragraph block.
- Press Shift + Tab to move focus to the block toolbar.
- Press the Right arrow key to move focus to the 'Move up' or 'Move down' button.
- Press Enter.
- Observe focus is unexpectedly moved to the first button in the toolbar.
- Expected: focus to stay on the button that was activated.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by annezazu** [comment]
@jeryj could you perhaps dig in here?

**Comment by t-hamano** [comment]
This issue does not occur in WordPress 6.4.3. Upon further investigation, it appears that this issue first occurred with Gutenberg plugin version 17.3.0.

I think what is shown in the video below is the expected behavior.


https://github.com/WordPress/gutenberg/assets/54422211/1412a958-4bcd-4f29-985e-40e2fa414ae9

**Comment by jeryj** [comment]
Looking at this today!

**Comment by jeryj** [comment]
Fix is ready in https://github.com/WordPress/gutenberg/pull/58570 along with tests to catch regression. Hopefully this addresses it and doesn't create anything new 😅
